src/jit/gentree.cpp

```diff
@@ -885,6 +885,42 @@ bool GenTreeCall::IsPure(Compiler* compiler) const
            compiler->s_helperCallProperties.IsPure(compiler->eeGetHelperNum(gtCallMethHnd));
 }
 
+bool GenTreeCall::IsSideEffectFree(Compiler* compiler, bool ignoreExceptions, bool ignoreCctors) const
+{
```

Comment:
1. Please add normal function header comment
2. Mention and call out relationship to gtNodeHasSideEffects() -- can that one use this one?
3. Should this instead be named HasSideEffects() to avoid the pseudo-double-negative?